src/ami-set.js

```diff
@@ -51,6 +55,49 @@ AmiSet.listAmiSets = async function () {
 };
 
 /**
+ * Checks if the AMIs from the amiSet are valid or not.
+ */
+
+async function checkAmi(ctx, region, ami, vtype) {
```

Comment:
Actually, I see why you're doing it now.  It's similar to the `createLaunchSpec` method, since we have to have a non-instance method for doing the check before creating and modifying, and Constructor methods as I suggested wouldn't have access to context.  I think that doing the following would match what we have elsewhere:

```javascript
//Helper method
AmiSet.__checkAmi = function (region, ami, vtype, ec2) { } 

// A constructor method available
AmiSet.validate = function (region, ami, vtype, ec2) { } 

// An instance copy
AmiSet.prototype.validate = function (region, ami, vtype) {
  return AmiSet.validate(region, ami, vtype, this.ec2);
} 

```